src/utils/canvas.ts

```diff
@@ -75,3 +75,11 @@ export async function getBlob(imageProperties: ImageProperties): Promise<Blob> {
     imgCanvas.remove();
   }
 }
+
+export async function validateCanvasDimensions(canvas: HTMLCanvasElement): Promise<boolean> {
```

Comment:
Looks great!

Maybe adding a comment that would explain why this check is needed would help future readers?

For example:
```javascript
// If the canvas exceeds the size limit for the browser, canvas.toBlob returns null.
// This function tests for this condition so that we can warn user.
```

Minor, fftr.

Reply:
I added just the first part, that's sufficient imo